### PR TITLE
added new postUpgradeCb check for CSCwe21982

### PR DESCRIPTION
**Pull request by welkin-he**
folks,
can you please review this pull request?
Thanks,
welkin

### CODE REVIEW

File: aci-preupgrade-validation-script.py [hunk 1]
```diff
@@ -3156,6 +3156,11 @@ def post_upgrade_cb_check(index, total_checks, cversion, tversion, **kwargs):
             "SinceVersion": "5.2(4d)",
             "Impact": "VLAN for missing Mo will not be deployed to leaf",
         },
+        "infraRsToInterfacePolProfileOpt": {
+            "CreatedBy": "infraRsToInterfacePolProfile",
+            "SinceVersion": ["5.2(8d)","6.0(3d)"],
```
Comment (monrog2) [inline aci-preupgrade-validation-script.py:3161]: better to pick one data type for `SinceVersion` and handle that. If `list` gives you the most flexibility, then convert the rest

File: aci-preupgrade-validation-script.py [hunk 2]
```diff
@@ -3167,12 +3172,21 @@ def post_upgrade_cb_check(index, total_checks, cversion, tversion, **kwargs):
         return POST
 
     for new_mo in new_mo_dict:
-        since_version = AciVersion(new_mo_dict[new_mo]['SinceVersion'])
-        created_by_mo = new_mo_dict[new_mo]['CreatedBy']
-
-        if since_version.newer_than(str(cversion)):
+        skip_current_mo=False
+        if isinstance(new_mo_dict[new_mo]['SinceVersion'],list):
```
Comment (monrog2) [inline aci-preupgrade-validation-script.py:3176]: If `list` gives you the most flexibility, convert the other entries to list and remove this check

**Review by monrog2 — CHANGES_REQUESTED** [review]

File: aci-preupgrade-validation-script.py (hunk 1 above)
Comment (welkin-he) [reply aci-preupgrade-validation-script.py:3161]: sounds good, change is done.

File: aci-preupgrade-validation-script.py (hunk 2 above)
Comment (welkin-he) [reply aci-preupgrade-validation-script.py]: sounds good, changed.

**Review by monrog2 — APPROVED** [review]

**Comment by monrog2** [comment]
Updated pytest output:
```
[Check  1/1] Post Upgrade Callback Integrity... Re-run script after APICs are upgraded and back to Fully-Fit  POST UPGRADE CHECK REQUIRED
[Check  1/1] Post Upgrade Callback Integrity... Re-run script after APICs are upgraded and back to Fully-Fit  POST UPGRADE CHECK REQUIRED
[Check  1/1] Post Upgrade Callback Integrity...                                                                                      PASS
[Check  1/1] Post Upgrade Callback Integrity...                                                                                      PASS
[Check  1/1] Post Upgrade Callback Integrity...                                                                   FAIL - OUTAGE WARNING!!
  Missed Objects       Impact
  --------------       ------
  infraImplicitSetPol  Infra implicit settings will not be deployed

  Recommended Action: Contact Cisco TAC with Output
  Reference Document: https://datacenter.github.io/ACI-Pre-Upgrade-Validation-Script/validations/#post-upgrade-callback-integrity


[Check  1/1] Post Upgrade Callback Integrity...                                                                                      PASS
[Check  1/1] Post Upgrade Callback Integrity...                                                                   FAIL - OUTAGE WARNING!!
  Missed Objects       Impact
  --------------       ------
  fvSlaDef             IPSLA monitor policy will not be deployed
  infraImplicitSetPol  Infra implicit settings will not be deployed

  Recommended Action: Contact Cisco TAC with Output
  Reference Document: https://datacenter.github.io/ACI-Pre-Upgrade-Validation-Script/validations/#post-upgrade-callback-integrity


[Check  1/1] Post Upgrade Callback Integrity...                                                                                      PASS
[Check  1/1] Post Upgrade Callback Integrity...                                                                   FAIL - OUTAGE WARNING!!
  Missed Objects                   Impact
  --------------                   ------
  fvSlaDef                         IPSLA monitor policy will not be deployed
  infraAssocEncapInstDef           VLAN for missing Mo will not be deployed to leaf
  infraImplicitSetPol              Infra implicit settings will not be deployed
  infraRsConnectivityProfileOpt    VPC for missing Mo will not be deployed to leaf
  infraRsToInterfacePolProfileOpt  VLAN for missing Mo will not be deployed to leaf

  Recommended Action: Contact Cisco TAC with Output
  Reference Document: https://datacenter.github.io/ACI-Pre-Upgrade-Validation-Script/validations/#post-upgrade-callback-integrity


[Check  1/1] Post Upgrade Callback Integrity...                                                                                      PASS
[Check  1/1] Post Upgrade Callback Integrity...                                                                   FAIL - OUTAGE WARNING!!
  Missed Objects                   Impact
  --------------                   ------
  compatSwitchHw                   Unexpected 64/32 bit image can deploy to switches
  fvSlaDef                         IPSLA monitor policy will not be deployed
  infraAssocEncapInstDef           VLAN for missing Mo will not be deployed to leaf
  infraImplicitSetPol              Infra implicit settings will not be deployed
  infraRsConnectivityProfileOpt    VPC for missing Mo will not be deployed to leaf
  infraRsToInterfacePolProfileOpt  VLAN for missing Mo will not be deployed to leaf

  Recommended Action: Contact Cisco TAC with Output
  Reference Document: https://datacenter.github.io/ACI-Pre-Upgrade-Validation-Script/validations/#post-upgrade-callback-integrity
```

**Review by monrog2 — APPROVED** [review]

**Pull request merged**